src/k8s.rs

```diff
@@ -1,13 +1,47 @@
+use std::fmt;
+
 use anyhow::Ok;
 use colored::Colorize;
 use futures_util::AsyncBufReadExt;
 use futures_util::TryStreamExt;
+use k8s_openapi::api::apps::v1::Deployment;
 use k8s_openapi::api::core::v1::Pod;
-use kube::api::{Api, LogParams};
+use kube::api::{Api, ListParams, LogParams};
+use kube::runtime::reflector::Lookup;
 use kube::ResourceExt;
 
+use itertools::Itertools;
+
 use crate::util;
 
+pub async fn get_pod_list_for_deployment(
+    client: &kube::Client,
+    deployment_name: &str,
+    ns_name: &str,
+) -> Result<Vec<String>, anyhow::Error> {
+    let deployment_api: Api<Deployment> = Api::namespaced(client.clone(), ns_name);
+    let deployment = deployment_api.get(deployment_name).await?;
+
+    let spec = deployment.spec.unwrap();
+    let match_labels = spec.selector.match_labels.unwrap();
+
+    let labels: String = match_labels
+        .iter()
+        .map(|(key, value)| format!("{}={}", key, value))
+        .join(",");
+
+    let pod_api: Api<Pod> = Api::namespaced(client.clone(), ns_name);
+    let list_params = ListParams::default().labels(&labels);
+    let pod_list = pod_api.list(&list_params).await?;
+
+    let mut pod_name_list: Vec<String> = vec!["".to_string(); 0];
```

Comment:
🚫 **[clippy]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> 
<pre><code>error: function or method calls as the initial value in zero-sized array initializers may cause side effects
  --> src/k8s.rs:37:5
   |
37 |     let mut pod_name_list: Vec<String> = vec!["".to_string(); 0];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `"".to_string(); let mut pod_name_list: std::vec::Vec<std::string::String> = vec![];`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#zero_repeat_side_effects
   = note: `-D clippy::zero-repeat-side-effects` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::zero_repeat_side_effects)]`
</code></pre>
<!-- __reviewdog__:ChBhZmNiZTQyOGMyYTllM2QwEgZjbGlwcHk= -->